crates/sui-bytecode-src-verifier/src/lib.rs

```diff
@@ -0,0 +1,253 @@
+        compiled_package: CompiledPackage,
+    ) -> Result<DependencyVerificationResult, DependencyVerificationError> {
+        let resolution_graph = match build_config.clone().resolution_graph_for_package(path) {
+            Ok(graph) => graph,
+            Err(err) => {
+                return Err(DependencyVerificationError::ResolutionGraphNotResolved(err));
+            }
+        };
+
+        let compiled_dep_map = Self::get_module_bytes_map(&compiled_package);
+
+        let mut on_chain_module_count = 0usize;
+        let mut verified_dependencies: HashMap<AccountAddress, Dependency> = HashMap::new();
+
+        for (pkg_symbol, resolution_package) in resolution_graph.package_table {
+            if pkg_symbol == compiled_package.compiled_package_info.package_name {
+                continue;
+            };
+
+            let local_pkg_bytes = match compiled_dep_map.get(&pkg_symbol) {
+                Some(module_bytes) => {
+                    if self.verbose {
+                        println!(
+                            "\nlocal package dependency {} : {} modules",
+                            pkg_symbol,
+                            module_bytes.len()
+                        );
+                    }
+                    module_bytes
+                }
+                None => {
+                    return Err(DependencyVerificationError::LocalDependencyNotFound(
+                        pkg_symbol, None,
+                    ));
+                }
+            };
+
+            for (symbol, addr) in resolution_package.resolution_table {
+                // package addresses may show up many times, but we only need to verify them once
+                // zero address is the package we're checking dependencies for
+                if verified_dependencies.contains_key(&addr) || addr.eq(&AccountAddress::ZERO) {
+                    continue;
+                }
+
+                // fetch the Sui object at the address specified for the package in the local resolution table
+                let on_chain_package = self.pkg_for_address(&addr).await?;
+
+                for (oc_name, on_chain_bytes) in &on_chain_package.module_map {
+                    let on_chain_module_symbol = Symbol::from(oc_name.as_str());
+                    let local_bytes = match local_pkg_bytes.get(&on_chain_module_symbol) {
+                        Some(bytes) => bytes,
+                        None => {
+                            return Err(DependencyVerificationError::LocalDependencyNotFound(
+                                pkg_symbol,
+                                Some(on_chain_module_symbol),
+                            ))
+                        }
+                    };
+
+                    // compare local bytecode to on-chain bytecode to ensure integrity of our dependencies
+                    if local_bytes != on_chain_bytes {
+                        return Err(DependencyVerificationError::ModuleBytecodeMismatch(
```

Comment:
I think showing the names of the module in question is probably good enough for now--it's hard to imagine the raw bytes being readable. Maybe a diff would be, though--curious for @awelc's thoughts after recently debugging a fine-grained bytecode mismatch.

Reply:
yeah, the raw bytes aren't super readable - I wasn't sure what level of detail we wanted to provide, so i defaulted to all of it.

Reply:
For now i've removed the bytes and just kept Package, module, address (could maybe even remove address)